### PR TITLE
Edit club, RankRequiredMixin and get_club_id helper

### PR DESCRIPTION
**Pull request by jc121F1**
Add RankRequiredMixin, get_club_id helper, and edit_club template.

Due to lack of sessions being set you cannot test the editing by hand yet. Understand if you wish to hold this merge till we have dashboard setting session.

### CODE REVIEW

File: BookClub/tests/helpers_tests/test_rank_required_mixin.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        self.url = reverse('test_rank_required')
+        self.user = User.objects.get(pk = 1)
+        self.club = Club.objects.get(pk=1)
+        return super().setUp()
+        
+    def test_user_has_required_rank_and_club(self):
+        self.client.login(username='johndoe', password='Password123')
+        self.assertEqual(self._is_logged_in(), True)
+        response = self.client.get(self.url)
+        self.assertEqual(response.url, "/login/")
```
Comment (clerav) [inline BookClub/tests/helpers_tests/test_rank_required_mixin.py:42]: use reverse here instead

Comment (jc121F1) [reply BookClub/tests/helpers_tests/test_rank_required_mixin.py:42]: In the assertion?

**Comment by jc121F1** [comment]
Closed as I'm not happy with this code yet.

File: BookClub/tests/helpers_tests/test_rank_required_mixin.py (hunk 1 above)
Comment (clerav) [reply BookClub/tests/helpers_tests/test_rank_required_mixin.py:42]: Yeah, to follow the same style in your other tests

**Pull request closed without merging**